Services/Mail/test/ilMailTest.php

```diff
@@ -177,4 +191,369 @@ static function (string $login) use ($loginToIdMap) : int {
 
         ilMimeMail::setDefaultTransport($oldTransport);
     }
+
+    public function testGetMailObjectReferenceId() : void
+    {
+        $refId = 364;
+        $instance = $this->create($refId);
+
+        $this->assertEquals($refId, $instance->getMailObjectReferenceId());
+    }
+
+    public function testFormatNamesForOutput() : void
+    {
+        $instance = $this->create();
+
+        $this->assertEquals('', $instance->formatNamesForOutput(''));
+        $this->assertEquals('', $instance->formatNamesForOutput(','));
+    }
+
+    /**
+     * @dataProvider provideGetPreviousMail
+     */
+    public function testGetPreviousMail(array $rowData) : void
+    {
+        $mailId = 3454;
+        $instance = $this->createAndExpectDatabaseCall($mailId, $rowData);
+        $this->mockDatabase->expects(self::once())->method('setLimit')->with(1, 0);
+        $instance->getPreviousMail($mailId);
+    }
+
+    public function provideGetPreviousMail() : array
+    {
+        return [
+            [[]],
+            [[
+                'allowed_classes' => '',
```

Comment:
`allowed_classes` is only used as an option parameter when unserializing data. Thy did you include this key here?